### PR TITLE
VACMS-8675: Port Behat tests to Cypress.

### LINKED ISSUE #8675: Create Cypress tests to replace existing Behat tests
## Description

There are Behat tests that need to be recreated in Cypress and removed from Behat.

## Acceptance Criteria
- [x] The below tests have been ported to Cypress with no loss of reliability or have been ported to another test system or deleted as appropriate.

## Implementation steps

- [tests/behat/features/content_editing.feature](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/tests/behat/features/content_editing.feature):
  - [x] Scenario: Log in and confirm that System-wide alerts can be created and edited
    - This is quite important.
    - This should be ported but we should improve its implementation.
  - [x] Scenario Outline: Confirm that content cannot be published directly from the node view but can from the node edit form.
    - This is probably a very, very slow test.
    - Think of how to refactor this to give a similar breadth of coverage more efficiently.
    - This might be done better as a unit test on a hook rather than a behavioral test.
  - [x] Scenario: Confirm Generate automatic URL alias is unchecked after node publish.
    - This is really a behavioral test that is confirming that a hook works as intended.
    - Rewrite as a unit test IMHO.
  - [x] Scenario: Confirm Generate automatic URL alias is unchecked after taxonomy term publish.
    - This should be a unit test IMHO.
  - [x] Confirm field group is opened and the required field is presented to editor on CLP page.
    - This might be reevaluated for utility.

- [tests/behat/features/decoupled.feature](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/tests/behat/features/decoupled.feature):
  - [x] Scenario: Log in, edit, publish, unpublish, and save nodes and see changes in the CMS and the front end website. Confirm cms menu items are in static site build.
    - Probably add some step definitions here.
    - Revisit scenario title and verify correctness in light of recent changes.

- [tests/behat/features/perms.feature](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/tests/behat/features/perms.feature):
  - [x] Scenario: Content Admins should be able to browse sections from their profile even if none have been specifically assigned to them
    - There're existing Cypress tests that cover adjacent functionality.  This should be a quick port, assuming it's not already present.

- [tests/behat/features/save_and_continue.feature](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/tests/behat/features/save_and_continue.feature):
  - [x] Scenario: Log in, edit, and save nodes with save and continue button and confirm revision saves changes.
    - This should be rephrased in terms of Cypress step definitions for brevity.

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Sitewide program`
- [x] `Platform CMS Team`
- [ ] `Sitewide crew ` (leave Sitewide unchecked and check the specific team instead)
  - [ ] `⭐️ Content ops`
  - [ ] `⭐️ CMS experience`
  - [ ] `⭐️ Public websites`
  - [ ] `⭐️ Facilities`
  - [ ] `⭐️ User support`

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Closes #8675.  The general goal here is to create a 1:1 copy for as many Behat tests as possible in Cypress and then delete the original Behat version.

## NEW 
- combines Cypress Accessibility and Behavioral tests into a single directory
- collects all `package.json` dev dependencies into `devDependencies`
- sets Cypress screenshots and videos to be deposited into docroot for easier viewing
- adds [`cypress/`](https://cms-a7hsns4nvftwek8nay2gakgoy3zdrzdr.ci.cms.va.gov/cypress/) web subdirectory with .htaccess to allow directory browsing

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As user _uid_ with _user_role_
1. Do this
   - [ ] Validate that
2. Then
   - [ ] Validate that
3. Then validate Acceptance Criteria from issue
   - [ ] a
   - [ ] b
   - [ ] c

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Sitewide program`
- [x] `Platform CMS Team`
- [x] `Sitewide crew ` (leave Sitewide unchecked and check the specific team instead)
  - [ ] `⭐️ Content ops`
  - [ ] `⭐️ CMS experience`
  - [ ] `⭐️ Public websites`
  - [ ] `⭐️ Facilities`
  - [ ] `⭐️ User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: tests/behavioral/cypress/support/commands.js [hunk 1]
```diff
@@ -67,11 +67,24 @@ Cypress.Commands.add('drupalDrushCommand', (command) => {
 });
 
 Cypress.Commands.add('drupalContentRelease', (command) => {
-  let cmd = 'composer va:web:build';
+  // cy.exec() starts at ${REPO_ROOT}/tests/behavioral
+  let cmd = 'cd ../.. && composer va:web:build';
   if (Cypress.env('VAGOV_INTERACTIVE')) {
-    cmd = 'ddev composer va:web:build';
+    // Remove once ddev frontend build works again.
+    cmd = 'echo ddev composer va:web:build';
```
Comment (ndouglas) [inline tests/behavioral/cypress/support/commands.js:74]: ddev frontend build is broken, so the only reason for performing this test locally is if you're actually developing the tests themselves.

File: tests/behavioral/cypress/support/commands.js [hunk 2]
```diff
@@ -67,11 +67,24 @@ Cypress.Commands.add('drupalDrushCommand', (command) => {
 });
 
 Cypress.Commands.add('drupalContentRelease', (command) => {
-  let cmd = 'composer va:web:build';
+  // cy.exec() starts at ${REPO_ROOT}/tests/behavioral
+  let cmd = 'cd ../.. && composer va:web:build';
   if (Cypress.env('VAGOV_INTERACTIVE')) {
-    cmd = 'ddev composer va:web:build';
+    // Remove once ddev frontend build works again.
```
Comment (ndouglas) [inline tests/behavioral/cypress/support/commands.js:73]:
```suggestion
    // Remove "echo" once ddev frontend build works again.
```

File: scripts/tests/cypress.sh [hunk 3]
```diff
@@ -11,12 +11,6 @@ set -e
 # Allow script to be run anywhere in git repo
 cd "$(git rev-parse --show-toplevel)"
 
-# Do not attempt to run on BRD environment.
-if [ -n "$CMS_IS_BRD" ]; then
-  echo "Detected BRD environment, skipping cypress-axe tests."
-  exit 0
-fi
-
```
Comment (ndouglas) [inline scripts/tests/cypress.sh:19]: BRD can now run Cypress tests 🎉

File: composer.json [hunk 4]
```diff
@@ -623,8 +623,11 @@
             "npm install",
             "npm run stylelint-themes"
         ],
-        "va:test:accessibility": [
-            "./scripts/tests/cypress.sh accessibility"
+        "va:test:cypress": [
+            "cd \"$(git rev-parse --show-toplevel)\"",
+            "[ -d node_modules ] || (echo \"Please run 'npm install' before running cypress tests\" && exit 1)",
+            "./node_modules/.bin/cypress install",
+            "npm run test:cypress"
```
Comment (ndouglas) [inline composer.json:630]: Replaces `scripts/tests/cypress.sh`.

File: cypress.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  "supportFile": "tests/cypress/support/index.js",
+  "testFiles": [
+    "behavioral/*.{feature,features}",
+    "accessibility/*.spec.js"
+  ],
+  "trashAssetsBeforeRuns": true,
+  "videoCompression": false,
+  "videoUploadOnPasses": false,
+  "viewportHeight": 900
+}
```
Comment (ndouglas) [inline cypress.json:22]: Moved from `tests/accessibility` and `tests/behavioral` for symmetry with phpunit, phpcs, etc.

File: package.json [hunk 6]
```diff
@@ -15,9 +15,8 @@
     "stylelint-modules": "npx stylelint 'docroot/modules/custom/**/*.css'",
     "stylelint-themes": "npx stylelint 'docroot/themes/custom/**/*.scss'",
     "stylelint": "npx stylelint",
-    "test:accessibility": "cd tests/accessibility && cypress run",
-    "test:behavioral": "cd tests/behavioral && cypress run -e \"TAGS=$CYPRESS_TAGS\"",
-    "test:interactive": "cd tests/behavioral && cypress open -e \"TAGS=not @ignore and not @piv,VAGOV_INTERACTIVE=true\""
+    "test:cypress": "cypress run -e \"TAGS=$CYPRESS_TAGS\"",
+    "test:cypress:interactive": "cypress open -e \"TAGS=not @ignore and not @piv,VAGOV_INTERACTIVE=true\""
```
Comment (ndouglas) [inline package.json:19]: Merged accessibility/behavioral.

File: package.json [hunk 7]
```diff
@@ -29,21 +28,7 @@
...
-    "cypress-axe": "^0.14.0",
-    "cypress-cucumber-preprocessor": "^4.3.1",
-    "cypress-real-events": "^1.6.0",
-    "debug": "^2.6.9",
-    "faker": "^5.3.1",
-    "pixelmatch": "^5.2.1",
-    "pngjs": "^6.0.0",
-    "selenium-webdriver": "^4.0.0"
-  },
+  "dependencies": {},
```
Comment (ndouglas) [inline package.json:31]: Was this too aggressive?  Will this cause problems?

File: tests/cypress/integration/accessibility/nodeCreationAccessibility.spec.js [hunk 8]
```diff
@@ -20,7 +20,7 @@ const routes = [
 before(() => {
   // @TODO Use Cypress.env variables for user/pass.
   // @TODO Use a content admin role.
-  cy.login('axcsd452ksey', 'drupal8');
+  cy.drupalLogin('axcsd452ksey', 'drupal8');
```
Comment (ndouglas) [inline tests/cypress/integration/accessibility/nodeCreationAccessibility.spec.js:23]: Reuse existing command from behavioral 😃

File: tests/cypress/integration/behavioral/common/i_create_a_node.js [hunk 9]
```diff
@@ -95,3 +127,22 @@ Given('I create a {string} node', (contentType) => {
...
+      const pathComponents = currentPath.split('/');
+      pathComponents.pop();
+      cy.wrap(pathComponents.pop()).as('nodeId');
+      cy.window().then((window) => {
+        const pagePath = window.location.pathname;
+        cy.wrap(pagePath).as('pagePath');
+      });
+    });
+  });
+});
```
Comment (ndouglas) [inline tests/cypress/integration/behavioral/common/i_create_a_node.js:148]: save and continue functionality

**Review by tjheffner — DISMISSED** [review]
i love this PR so much. thank you for making the cypress videos easily accessible

File: docroot/sites/default/settings/settings.staging.php [hunk 10]
```diff
@@ -35,6 +35,7 @@
     '10\.247.*',
     'localhost',
     'va-gov-cms.lndo.site',
+    'va-gov-cms.ddev.site',
```
Comment (ndouglas) [inline docroot/sites/default/settings/settings.staging.php:38]: Some tests depended on this being set prior to https://github.com/department-of-veterans-affairs/devops/pull/11104

File: docroot/sites/default/settings/settings.prod.php [hunk 11]
```diff
@@ -35,6 +35,7 @@
     '10\.247.*',
     'localhost',
     'va-gov-cms.lndo.site',
+    'va-gov-cms.ddev.site',
```
Comment (ndouglas) [inline docroot/sites/default/settings/settings.prod.php:38]: Some tests depended on this being set prior to https://github.com/department-of-veterans-affairs/devops/pull/11104

File: docroot/sites/default/settings/settings.dev.php [hunk 12]
```diff
@@ -37,6 +37,7 @@
     'dev.cms.va.gov',
     'test.dev.cms.va.gov',
     '.*\.us-gov-west-1\.elb\.amazonaws\.com',
+    'va-gov-cms.ddev.site',
```
Comment (ndouglas) [inline docroot/sites/default/settings/settings.dev.php:40]: Some tests depended on this being set prior to https://github.com/department-of-veterans-affairs/devops/pull/11104

File: docroot/cypress/.htaccess [hunk 13]
```diff
@@ -0,0 +1 @@
+Options +Indexes
```
Comment (ndouglas) [inline docroot/cypress/.htaccess:1]: Provide easier access to screenshots and videos captured by Cypress tests.

File: composer.json [hunk 14]
```diff
@@ -623,11 +623,11 @@
...
-        "va:test:accessibility": [
-            "./scripts/tests/cypress.sh accessibility"
-        ],
-        "va:test:behavioral": [
-            "./scripts/tests/cypress.sh behavioral"
+        "va:test:cypress": [
+            "cd \"$(git rev-parse --show-toplevel)\"",
+            "[ -d node_modules ] || npm install",
+            "./node_modules/.bin/cypress install",
+            "npm run test:cypress"
```
Comment (ndouglas) [inline composer.json:630]: Replaces the `scripts/tests/cypress.sh` script.

**Review by cweagans — APPROVED** [review]
This looks great. Very excited to get this merged :)

**Review by swirtSJW — COMMENTED** [review]
@ndouglas This is pretty amazing work.
Only one concern.  We lost a Readme and no documentation was added.

**Review by swirtSJW — APPROVED** [review]
I got a pinky swear that documentation update is coming.

**Pull request merged**